### PR TITLE
use built-in idna instead of thrid party

### PR DESCRIPTION
**Pull request by karpetrosyan**
# Summary

Use built-in idna instead of third party library.

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

**Comment by karosis88** [comment]
Thinking about why the output is not the same.

``` python
import encodings.idna

print('Königsgäßchen.de'.encode('idna'), encodings.idna.ToASCII('Königsgäßchen.de').decode('ascii'))
```

out:

`.encode` b'xn--knigsgsschen-lcb0w.de' 
`encodings.idna.ToASCII` xn--knigsgsschen.de-5kb41a

File: httpx/_urlparse.py [hunk 1]
```diff
@@ -324,8 +322,8 @@ def encode_host(host: str) -> str:
 
     # IDNA hostnames
     try:
-        return idna.encode(host.lower()).decode("ascii")
-    except idna.IDNAError:
+        return host.lower().encode("idna").decode("ascii")
+    except Exception:
```
Comment (karosis88) [inline httpx/_urlparse.py:326]:
```suggestion
    except UnicodeError:
```

**Comment by T-256** [comment]
perhaps we could still support 3dparty idna as optional dependency.

**Comment by jruere** [comment]
@karpetrosyan, the built-in IDNA encoding implements an outdated standard. It should not be used.

See the RFC each implement:
- https://docs.python.org/3/library/codecs.html#text-encodings
- https://pypi.org/project/idna/ (also explains this difference)

AFAIU, Python will not update the encoding as it would be a breaking change.

**Pull request closed without merging**